lib/ansible/module_utils/vmware.py

```diff
@@ -29,20 +30,42 @@ class TaskError(Exception):
     pass
 
 
-def wait_for_task(task):
+def wait_for_task(task, max_backoff=64, timeout=3600):
+    """Wait for given task using exponential back-off algorithm.
+
+    Args:
+        task: VMware task object
+        max_backoff: Maximum amount of sleep time in seconds
+        timeout: Timeout for the given task in seconds
+
+    Returns: True with result for successful task.
+    Raises: TaskError on failure
+    """
+    failure_counter = 0
+    start_time = time.time()
+
+    # List of generic errors
+    gen_errors = (
+        vim.fault.CannotAccessVmConfig,
+        vmodl.fault.InvalidArgument,
+    )
 
     while True:
+        if time.time() - start_time >= timeout:
+            raise TaskError("Timeout")
         if task.info.state == vim.TaskInfo.State.success:
             return True, task.info.result
         if task.info.state == vim.TaskInfo.State.error:
             try:
+                if isinstance(task.info.error, gen_errors):
```

Comment:
Also, it looks like you could stop relying on exception comparison and  do smth like:
```python
from ansible.module_utils import six
"""Do smth like this in module heading."""
...
error_msg = task.info.error
try:
    error_msg = error_msg.msg
except AttributeError:
    """Keep message as is if it doesn't have `msg` attribute."""
    pass
finally:
    """
    six's `raise_from(exception1, exeption2)` does `raise exception1 from exeption2` under Python 3 where it's supported and will do `raise exception1`, where it's not. 
    """
    six.raise_from(TaskError(error_msg), task.info.error)
```